test/WebJobs.Script.Tests.Shared/TestHelpers.cs

```diff
@@ -35,7 +35,9 @@ public static string FunctionsTestDirectory
         {
             get
             {
-                return Path.Combine(Path.GetTempPath(), "FunctionsTest");
+                var path = Path.Combine(Path.GetTempPath(), "FunctionsTest");
```

Comment:
This does change the behavior of this helper method. Are you sure there's no existing code that expects to test the result of this property? You may also want to consider changing this to a method and make its behavior clear in the name (as it isn't after this change). That could have a larger impact on the changes. Adding a method to this type that introduces this logic would be a better option.

Reply:
Have reverted this and updated the logic in the metadata provider to be a bit more resilient